apps/previewgenerator.sh

```diff
@@ -16,67 +16,258 @@ debug_mode
 # Check if root
 root_check
 
-msg_box "Please note that this will put your server at risk as it will install a package called 'imagick' which is known to have several flaws.
+# PHP 7.x is needed
+if is_this_installed php5.6-common
+then
+    msg_box "At least PHP 7.X is supported, please upgrade your PHP version: https://shop.hanssonit.se/product/upgrade-php-version-including-dependencies/"
+    exit
+elif is_this_installed php5.5-common
+then
+    msg_box "At least PHP 7.X is supported, please upgrade your PHP version: https://shop.hanssonit.se/product/upgrade-php-version-including-dependencies/"
+    exit
+fi
 
-You can check this issue to understand why: https://github.com/nextcloud/vm/issues/743
+# Encryption may not be enabled #### encryption will always list, check if it's enabled by grepping the version number
+#if occ_command app:list | grep encryption
+#then
+#    msg_box "It seems like you have encryption enabled which is unsupported when using the preview generator"
+#    exit
+#fi
+
+msg_box "This script will install the previewgerator. 
 
-You can choose to cancel installing this in the next step."
-if [[ "no" == $(ask_yes_or_no "Do you still want to continue?") ]]
+It can speedup the loading of previews in Nextcloud a lot.
+
+Please note: If you continue, all your current previewgenerator settings will be lost, if any."
+if [[ "yes" == $(ask_yes_or_no "Do you want to install the previewgenerator?") ]]
 then
-    exit
+    # Install preview generator
+    echo "install the preview generator"
+    install_and_enable_app previewgenerator
+    
+    # check if the previewgenerator is installed and enabled
+    if [ -d "$NC_APPS_PATH/previewgenerator" ]
+    then
+        # enable previews
+        occ_command config:system:set enable_previews --value=true --type=boolean
+        
+        # install needed dependency for movies
+        install_if_not ffmpeg
+        
+        # reset the preview formats
+        occ_command config:system:delete "enabledPreviewProviders"
+        
+        # reset the cronjob
+        print_text_in_color "$ICyan" "Resetting the cronjob for the preview-generation"
+        crontab -u www-data -l | grep -v 'preview:generate-all'  | crontab -u www-data -
```

Comment:
@enoch85 this doesn't remove anything in the moment.
do you have another idea what to change to make it work?

Reply:
When I tested it, it removed everything. Using `sed` isn't a clean solution imo.

Reply:
Hm...

Reply:
It works as expected.

Reply:
Okay